packages/server/src/fhir/operations/agentbulkstatus.ts

```diff
@@ -0,0 +1,73 @@
+import { allOk, badRequest, isOk, serverError } from '@medplum/core';
+import { FhirRequest, FhirResponse } from '@medplum/fhir-router';
+import { Agent, Bundle, BundleEntry, OperationDefinition, OperationOutcome, Parameters } from '@medplum/fhirtypes';
+import { getAuthenticatedContext } from '../../context';
+import { agentStatusHandler } from './agentstatus';
+import { getAgentsForRequest } from './agentutils';
+
+export const operation: OperationDefinition = {
+  resourceType: 'OperationDefinition',
+  name: 'agent-bulk-status',
+  status: 'active',
+  kind: 'operation',
+  code: 'bulk-status',
+  experimental: true,
+  resource: ['Agent'],
+  system: false,
+  type: true,
+  instance: false,
+  parameter: [{ use: 'out', name: 'return', type: 'Bundle', min: 1, max: '1' }],
+};
+
+/**
+ * Handles HTTP requests for the Agent $status operation.
+ * First reads the agent and makes sure it is valid and the user has access to it.
+ * Then tries to get the agent status from Redis.
+ * Returns the agent status details as a Parameters resource.
+ *
+ * @param req - The FHIR request.
+ * @returns The FHIR response.
+ */
+export async function agentBulkStatusHandler(req: FhirRequest): Promise<FhirResponse> {
+  const { repo } = getAuthenticatedContext();
+  const agents = await getAgentsForRequest(req, repo);
```

Comment:
Low pri: should we consider adding an upper bound on number of agents?

Pagination should "just work" with your current implementation, because `_count` and `_offset` should flow through automatically.

I'd be ok with a reasonably high limit here, like 100